File: app/src/main/java/com/letianpai/robot/desktop/storage/RobotConfigManager.java
package com.letianpai.robot.desktop.storage;

import android.content.Context;



/**
 * 机器人 偏好设置管理器
 * @author liujunbin
 */
public class RobotConfigManager implements RobotConfigConst {

    private static RobotConfigManager mRobotConfigManager;
    private RobotSharedPreference mRobotSharedPreference;
    private Context mContext;


    private RobotConfigManager(Context context) {
        this.mContext = context;
        this.mRobotSharedPreference = new RobotSharedPreference(context,
                RobotSharedPreference.SHARE_PREFERENCE_NAME, RobotSharedPreference.ACTION_INTENT_CONFIG_CHANGE);
    }


    private void initKidSmartConfigState() {

    }

    public static RobotConfigManager getInstance(Context context) {
        if (mRobotConfigManager == null) {
            mRobotConfigManager = new RobotConfigManager(context);
            mRobotConfigManager.initKidSmartConfigState();
            mRobotConfigManager.commit();
        }
        return mRobotConfigManager;

    }

    public boolean commit() {
        return mRobotSharedPreference.commit();
    }



    public void setAppListConfig(String appList){
        mRobotSharedPreference.putString(KEY_APP_LIST_CONFIG,appList);
    }

    public String getAppListConfig(){
        return mRobotSharedPreference.getString(KEY_APP_LIST_CONFIG,VALUE_DEFAULT_APP_LIST_CONFIG);
    }



}

File: app/src/main/java/com/letianpai/robot/desktop/storage/RobotConfigManager.kt
package com.letianpai.robot.desktop.storage

import android.content.Context

/**
 * Robotics Preference Manager
 * @author liujunbin
 */
class RobotConfigManager private constructor(private val mContext: Context) : RobotConfigConst {
    private val mRobotSharedPreference =
        RobotSharedPreference(
            mContext,
            RobotSharedPreference.SHARE_PREFERENCE_NAME,
            RobotSharedPreference.ACTION_INTENT_CONFIG_CHANGE
        )


    private fun initKidSmartConfigState() {
    }

    fun commit(): Boolean {
        return mRobotSharedPreference.commit()
    }


    var appListConfig: String?
        get() = mRobotSharedPreference.getString(
            RobotConfigConst.KEY_APP_LIST_CONFIG,
            RobotConfigConst.VALUE_DEFAULT_APP_LIST_CONFIG
        )
        set(appList) {
            mRobotSharedPreference.putString(
                RobotConfigConst.KEY_APP_LIST_CONFIG,
                appList!!
            )
        }


    companion object {
        private var mRobotConfigManager: RobotConfigManager? = null
        fun getInstance(context: Context): RobotConfigManager? {
            if (mRobotConfigManager == null) {
                mRobotConfigManager = RobotConfigManager(context)
                mRobotConfigManager!!.initKidSmartConfigState()
                mRobotConfigManager!!.commit()
            }
            return mRobotConfigManager
        }
    }
}
